R/familiasTodvir.R

```diff
@@ -52,40 +60,63 @@ familiasTodvir = function(famfile, missingPrefix = "M", verbose = FALSE){
       ref
     })
   }
+  
   # Check for identically named reference individuals
   if(!is.null(am)){
     if(any(duplicated(typedMembers(am))))
       stop2("Typed members of reference families must be named differently.")
   }
-  # Treat cases with one reference family and several separately
-  if (is.ped(am)){
-    missing = grep("^Missing", unlist(labels(am)), value = TRUE)
-    m = length(missing)
-    if(m == 0)
-      stop2("Reference pedigree without missing.")
-    else{
-      newMissing = paste(missingPrefix, 1:m, sep = ".")
-      am = relabel(am, new = newMissing, old =  missing)
-    }
-    missing = newMissing
-  }
-  else if (is.pedList(am)){
-    missing = NULL
-    j = 1
-    for (i in 1:length(am)){
-      missing1 = grep("^Missing", unlist(labels(am[[i]])), value = TRUE)
-      m = length(missing1)
-      if(m == 0)
-        stop2("Reference pedigree without missing.")
-      else{
-        for (k in 1:m){
-          newMissing = paste(missingPrefix,j, sep = ".",k)
-          am[[i]] = relabel(am[[i]], newMissing, missing1[k])
-          missing = c(missing, newMissing)
-        }
-        j = j + m
-      }
-    }
-  }
-    dviData(pm = pm, am = am, missing = missing)
+  
+  
+  # Missing individuals -----------------------------------------------------
+
+  missing = grep("^Missing", unlist(labels(am)), value = TRUE)
+  
+  if(!length(missing))
+    stop2("Reference pedigree without missing")
+  
+
+  # Create DVI object -------------------------------------------------------
+
+  dvi0 = dviData(pm = pm, am = am, missing = missing)
+  
+  # Relabel missing persons
+  if(is.ped(am))
+    dvi = relabelDVI(dvi0, missingPrefix = missingPrefix)
+  else
+    dvi = relabelDVI(dvi0, missingFormat = "M.[FAM].[IDX]")
+  
+  dvi
+  
```

Comment:
I suggest letting `relabelDVI()` take care of the relabelling. The new parameter `missingFormat` is quite flexible!

Reply:
I modified the code using the new `relabelDVI()`. Can you test it?

Reply:
Great improvements @magnusdv ! I have incorporated also the changes you only suggested (removed line 48, simplified `lapply`).. Using `relabelDVI` is very efficient and I added the other arguments of this function, very useful since sometimes people use names that need to be changed to make output and plots inteligible.